### PR TITLE
Bump protobuf to v25.3

### PR DESCRIPTION
**Pull request by Avasam**
Follow-up to #11693
~~Do not merge until `types-protobuf==4.24` has been published to PyPI !~~

Bumped mypy-protobuf to 3.6
Ran stub defaulter.

### CODE REVIEW

File: scripts/sync_tensorflow_protobuf_stubs.sh [hunk 1]
```diff
@@ -61,7 +61,6 @@ rm tensorflow/compiler/xla/service/hlo_execution_profile_data_pb2.pyi \
    tensorflow/core/protobuf/worker_service_pb2.pyi \
    tensorflow/core/util/example_proto_fast_parsing_test_pb2.pyi
```
Comment (Avasam) [inline scripts/sync_tensorflow_protobuf_stubs.sh:63]:
```suggestion


```

File: scripts/generate_proto_stubs.sh [hunk 2]
```diff
@@ -79,7 +79,4 @@ sed --in-place="" \
 cd $REPO_ROOT
 # use `|| true` so the script still continues even if a pre-commit hook
 # applies autofixes (which will result in a nonzero exit code)
-pre-commit run --files $(git ls-files -- "$REPO_ROOT/stubs/protobuf") || true
-# Ruff takes two passes to fix everything, re-running all of pre-commit is *slow*
-# and we don't need --unsafe-fixes to remove imports
-ruff check "$REPO_ROOT/stubs/protobuf" --fix --exit-zero
+pre-commit run --files $(git ls-files -- "$REPO_ROOT/stubs/protobuf/**.pyi") || true
```
Comment (Avasam) [inline scripts/generate_proto_stubs.sh:82]:
```suggestion
pre-commit run --files $(git ls-files -- "$REPO_ROOT/stubs/protobuf/**_pb2.pyi") || true
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -13,14 +13,15 @@ repos:
...
-        args: ["--exit-non-zero-on-fix"]
       - id: ruff
         # Run this separately because we don't really want
         # to use --unsafe-fixes for all rules
+        # Should be run first as it can leave unused imports behind
         name: Remove unnecessary `sys.version_info` blocks
         args: ["--exit-non-zero-on-fix", "--select=UP036", "--unsafe-fixes"]
+      - id: ruff
+        name: Run ruff on stubs, tests and scripts
+        args: ["--exit-non-zero-on-fix"]
```
Comment (srittau) [inline .pre-commit-config.yaml:24]: Could you move this to an separate PR, for cleaner separation of concerns?

Comment (Avasam) [reply .pre-commit-config.yaml:24]: Extracted to https://github.com/python/typeshed/pull/11737

Comment (Avasam) [inline .pre-commit-config.yaml:24]:
```suggestion
        name: Run ruff on stubs, tests and scripts
        args: ["--exit-non-zero-on-fix"]
      - id: ruff
        # Run this separately because we don't really want
        # to use --unsafe-fixes for all rules
        name: Remove unnecessary `sys.version_info` blocks
        args: ["--exit-non-zero-on-fix", "--select=UP036", "--unsafe-fixes"]
```

Reverting changes in this file, moved to https://github.com/python/typeshed/pull/11737

File: scripts/generate_proto_stubs.sh [hunk 4]
```diff
@@ -79,7 +79,4 @@ sed --in-place="" \
 cd $REPO_ROOT
```
Comment (Avasam) [inline scripts/generate_proto_stubs.sh:79]:
```suggestion
cd "$REPO_ROOT"
```

**Review by srittau — APPROVED** [review]
